### PR TITLE
Fix a bug where RPC calls before proxy finishes init crashes Milvus

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind bug

issue: #16473

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

**Comment by soothing-rain** [comment]
rerun ut

File: internal/proxy/authentication_interceptor.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ func AuthenticationInterceptor(ctx context.Context) (context.Context, error) {
 	if !ok {
 		return nil, ErrMissingMetadata()
 	}
+	if globalMetaCache == nil {
+		return nil, ErrGlobalMetaCacheUninitialized()
```
Comment (congqixia) [inline internal/proxy/authentication_interceptor.go:66]: NIT: Error message here is not user friendly. Actual case was calling a grpc before proxy is healthy.

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/proxy/authentication_interceptor.go (hunk 1 above)
Comment (soothing-rain) [reply internal/proxy/authentication_interceptor.go:66]: Yes. I will instead return errors like "proxy not initialized" or "proxy is not ready".

**Pull request merged**